torch/_functorch/_aot_autograd/dispatch_and_compile_graph.py

```diff
@@ -291,6 +298,27 @@ def aot_dispatch_autograd_graph(
 
     fx_g = _create_graph(joint_fn_to_trace, updated_joint_inputs, aot_config=aot_config)
 
+    # See Note [Side-Effectful Tokens in AOTAutograd]
+    # If forward graph does not use effectful ops with Tokens, we have not added additional token arguments before final tracing.
+    # If backward graph does use effectful ops with Tokens,
+    # we have to add additional token argument to the joint graph manually after tracing.
+    if fw_metadata.num_backward_discovered_tokens > 0:
+        add_discovered_token_in_backward_as_input(
+            fx_g, fw_metadata.num_backward_discovered_tokens
+        )
+        if aot_config.enable_log:
+            aot_graphs_effects_log.debug(
+                "%s",
+                lazy_format_graph_code(
+                    "aot_config id: %s, fw_metadata=%s, Joint graph after adding effect tokens discovered in backward",
```

Comment:
tbh, I don't think that the log is necessary, since the existing `aot_graphs` logging will log the final joint and fw/bw graphs after effect tokens have been added anyway (also, this won't matter if we add the extra effect tokens are *during* tracing, from the other comment)